### PR TITLE
Allow configuring jails with PTP network interfaces

### LINKED ISSUE #1041: IP Validation blocks PTP interfaces
PR #792 added validation for IP addresses.  However, it wrongly excluded assigning point-to-point addresses to jails.  The below commands worked on earlier versions of iocage, and resulted in a jail that directed all of its traffic over the ptp interface.  On the current version of iocage, this command fails.  Commenting out the check added in #792 fixes the problem.

```
$ iocage --version
Version	1.1 RELEASE 2019/01
$ iocage set ip4_addr="tun256|192.168.154.97 192.168.154.1" vpn
Please provide a valid ip: Expected 4 octets in '192.168.154.97 192.168.154.1'
```

### PR DESCRIPTION
**Pull request by asomers**
For example:
iocage set ip4_addr="tun256|192.168.154.97 192.168.154.1" vpn

Fixes #1041

Make sure to follow and check these boxes before submitting a PR! Thank you.

- [x] Explain the feature
- [x] Read [CONTRIBUTING.md](https://github.com/iocage/iocage/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by asomers** [comment]
ping.  Can somebody please review this PR?

**Comment by asomers** [comment]
ping.  Can somebody please review?

File: tests/unit_tests/1004_lib_ip_validation_test.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+def test_01_ip4_addr():
+    IOCJson.validate_ip4_addr('192.0.0.1')
+
+def test_01_ip4_net():
+    IOCJson.validate_ip4_addr('192.0.0.1/24')
+
+def test_01_ip4_ptp():
+    IOCJson.validate_ip4_addr('192.0.0.100/32 192.0.0.1')
+
+def test_01_ip6_addr():
```
Comment (sonicaj) [inline tests/unit_tests/1004_lib_ip_validation_test.py:15]: Can you please bump test numbers ?

File: iocage_lib/ioc_json.py [hunk 2]
```diff
@@ -2725,3 +2724,23 @@ def json_migrate_uuid_to_tag(self, uuid, tag, state, conf):
...
         return (conf, False, True)
+
+    @staticmethod
+    def validate_ip4_addr(ip):
+        # PTP configurations have two addresses.  Otherwise there will be 1.
+        parts = ip.split(" ")
+        if len(parts) > 2:
+            raise ValueError("Unrecognized IP4 address format")
+        if len(parts) == 2:
+            ipaddress.IPv4Address(parts[1])
```
Comment (sonicaj) [inline iocage_lib/ioc_json.py:2735]: Do we want `strict=False` here as well ? ( same for ipv6 below )

**Review by sonicaj — COMMENTED** [review]
Thank you @asomers, a couple of minor comments

**Comment by skarekrow** [comment]
Sorry about that! Some git-fu backfired, can you please reopen this? Thanks!

**Pull request closed without merging**